juju/juju
# Move caasapplicationprovisioner facade to use model services

The `ModelConfig` and `WatchForModelConfigChanges` methods of `caasapplicationprovisioner.Model` have been replaced by a model config service:

```go
type ModelConfigService interface {
	ModelConfig(context.Context) (*config.Config, error)
	Watch() (corewatcher.StringsWatcher, error)
}
```

Likewise, the `UUID` method of `caasapplicationprovisioner.Model` has been replaced by the model info service.

All services (including the existing ControllerConfigService) have been moved into a service.go file. I have modified the unit tests in this package to use gomock generated mocks for all of these services.

As a drive-by, I removed a line from the `smoke_k8s` test to fix the test. That line was asking for user confirmation, which is problematic in an automated test. I could have added `--no-prompt`, however it turns out that line is no longer necessary since the corresponding teardown bug has since been fixed.

## Checklist




## QA steps

Run unit tests. Run integration tests (`smoke_k8s`).

Try to deploy cos-lite:
```
juju bootstrap minikube c
juju add-model m
juju deploy cos-lite --trust
```

Check in `juju status` that the applications have come up correctly. Check in `juju debug-log` that there are no provisioning errors.

## Links

**Jira card:** JUJU-6054
